ospfd/ospf_vty.c

```diff
@@ -5805,41 +5872,97 @@ static const char *const show_database_header[] = {
 	"Opaque-Type/Id  ADV Router      Age  Seq#       CkSum",
 };
 
-static void show_ip_ospf_database_header(struct vty *vty, struct ospf_lsa *lsa)
+static void show_ip_ospf_database_header(struct vty *vty, struct ospf_lsa *lsa,
+					 json_object *json)
 {
 	struct router_lsa *rlsa = (struct router_lsa *)lsa->data;
 
-	vty_out(vty, "  LS age: %d\n", LS_AGE(lsa));
-	vty_out(vty, "  Options: 0x%-2x : %s\n", lsa->data->options,
-		ospf_options_dump(lsa->data->options));
-	vty_out(vty, "  LS Flags: 0x%-2x %s\n", lsa->flags,
-		((lsa->flags & OSPF_LSA_LOCAL_XLT) ? "(Translated from Type-7)"
-						   : ""));
-
-	if (lsa->data->type == OSPF_ROUTER_LSA) {
-		vty_out(vty, "  Flags: 0x%x", rlsa->flags);
-
-		if (rlsa->flags)
-			vty_out(vty, " :%s%s%s%s",
-				IS_ROUTER_LSA_BORDER(rlsa) ? " ABR" : "",
-				IS_ROUTER_LSA_EXTERNAL(rlsa) ? " ASBR" : "",
-				IS_ROUTER_LSA_VIRTUAL(rlsa) ? " VL-endpoint"
-							    : "",
-				IS_ROUTER_LSA_SHORTCUT(rlsa) ? " Shortcut"
-							     : "");
+	if (!json) {
+		vty_out(vty, "  LS age: %d\n", LS_AGE(lsa));
+		vty_out(vty, "  Options: 0x%-2x : %s\n", lsa->data->options,
+			ospf_options_dump(lsa->data->options));
+		vty_out(vty, "  LS Flags: 0x%-2x %s\n", lsa->flags,
+			((lsa->flags & OSPF_LSA_LOCAL_XLT)
+				 ? "(Translated from Type-7)"
+				 : ""));
+
+		if (lsa->data->type == OSPF_ROUTER_LSA) {
+			vty_out(vty, "  Flags: 0x%x", rlsa->flags);
+
+			if (rlsa->flags)
+				vty_out(vty, " :%s%s%s%s",
+					IS_ROUTER_LSA_BORDER(rlsa) ? " ABR"
+								   : "",
+					IS_ROUTER_LSA_EXTERNAL(rlsa) ? " ASBR"
+								     : "",
+					IS_ROUTER_LSA_VIRTUAL(rlsa)
+						? " VL-endpoint"
+						: "",
+					IS_ROUTER_LSA_SHORTCUT(rlsa)
+						? " Shortcut"
+						: "");
 
-		vty_out(vty, "\n");
+			vty_out(vty, "\n");
+		}
+		vty_out(vty, "  LS Type: %s\n",
+			lookup_msg(ospf_lsa_type_msg, lsa->data->type, NULL));
+		vty_out(vty, "  Link State ID: %s %s\n",
+			inet_ntoa(lsa->data->id),
+			lookup_msg(ospf_link_state_id_type_msg, lsa->data->type,
+				   NULL));
+		vty_out(vty, "  Advertising Router: %s\n",
+			inet_ntoa(lsa->data->adv_router));
+		vty_out(vty, "  LS Seq Number: %08lx\n",
+			(unsigned long)ntohl(lsa->data->ls_seqnum));
+		vty_out(vty, "  Checksum: 0x%04x\n",
+			ntohs(lsa->data->checksum));
+		vty_out(vty, "  Length: %d\n\n", ntohs(lsa->data->length));
+	} else {
+		char seqnum[10];
+		char checksum[10];
+
+		snprintf(seqnum, 10, "%x", ntohl(lsa->data->ls_seqnum));
+		snprintf(checksum, 10, "%x", ntohs(lsa->data->checksum));
+
+		json_object_int_add(json, "lsAge", LS_AGE(lsa));
+		json_object_string_add(json, "options",
```

Comment:
replace lsAge by lsaAge, because you have used lsaAge at other palces.

Reply:
Done